src/common/datetime.cpp

```diff
@@ -2271,6 +2272,142 @@ size_t wxDateTimeWorkDays::DoGetHolidaysInRange(const wxDateTime& dtStart,
+    const int days = (int)(22 + D + E);
+
+    // A corner case,
+    // when D is 29
+    if ((D == 29) && (E == 6))
+    {
+        wxASSERT_MSG(
+            wxDateTime(19, wxDateTime::Apr, year).GetWeekDay() ==
+            wxDateTime::WeekDay::Sun,
+            "Error in Easter calculation!");
+        return wxDateTime(19, wxDateTime::Apr, year);
+    }
+    // Another corner case,
+    // when D is 28
+    else if ((D == 28) && (E == 6))
+    {
+        wxASSERT_MSG(
+            wxDateTime(18, wxDateTime::Apr, year).GetWeekDay() ==
+            wxDateTime::WeekDay::Sun,
+            "Error in Easter calculation!");
+        return wxDateTime(18, wxDateTime::Apr, year);
+    }
+    else {
+        // If days > 31, move to April
+        // April = 4th Month
+        if (days > 31)
+        {
+            wxASSERT_MSG(
+                wxDateTime((days - 31), wxDateTime::Apr, year).GetWeekDay() ==
+                wxDateTime::WeekDay::Sun,
+                "Error in Easter calculation!");
+            return wxDateTime((days - 31), wxDateTime::Apr, year);
+        }
+        else
+        {
+            // Otherwise, stay on March
+            // March = 3rd Month
+            wxASSERT_MSG(
+                wxDateTime(days, wxDateTime::Mar, year).GetWeekDay() ==
+                wxDateTime::WeekDay::Sun,
+                "Error in Easter calculation!");
+            return wxDateTime(days, wxDateTime::Mar, year);
+        }
+    }
+}
+
+size_t wxDateTimeUSCatholicFeasts::DoGetHolidaysInRange(const wxDateTime& dtStart,
+                                                          const wxDateTime& dtEnd,
+                                                          wxDateTimeArray& holidays) const
+{
+    holidays.Clear();
+
+    for (wxDateTime dt = dtStart; dt <= dtEnd; dt += wxDateSpan::Day())
+    {
+        if (dt.IsSameDate(GetEaster(dt.GetYear())) ||
+            dt.IsSameDate(GetThursdayAscension(dt.GetYear())))
+        {
+            holidays.Add(dt);
+            continue;
+        }
+        for (const auto& holiday : m_holyDaysOfObligation)
+        {
+            if (holiday.GetMonth() == dt.GetMonth() &&
+                holiday.GetDay() == dt.GetDay())
+            {
+                holidays.Add(dt);
+            }
+        }
+    }
+
+    return holidays.size();
+}
+
+// ----------------------------------------------------------------------------
+// wxDateTimeChristianHolidays
+// ----------------------------------------------------------------------------
+
+size_t wxDateTimeChristianHolidays::DoGetHolidaysInRange(const wxDateTime& dtStart,
+                                                         const wxDateTime& dtEnd,
+                                                         wxDateTimeArray& holidays) const
+{
+    holidays.Clear();
+
+    for (wxDateTime dt = dtStart; dt <= dtEnd; dt += wxDateSpan::Day())
+    {
+        if (dt.IsSameDate(GetEaster(dt.GetYear())) ||
+            (dt.GetMonth() == 12 && dt.GetDay() == 25))
```

Comment:
Is there any reason not to use `DoIsHoliday()` here? I could understand it if we optimized this by precomputing Easter date only once outside of the loop, but we don't even do this...

Reply:
Got it, fixed it in both classes.